pkg/router/controller/status.go

```diff
@@ -133,16 +139,39 @@ func (a *StatusAdmitter) RecordRouteRejection(route *routev1.Route, reason, mess
 // RecordRouteUnservableInFutureVersions attempts to update the route status with a
 // reason for a route being unservable in future versions.
 func (a *StatusAdmitter) RecordRouteUnservableInFutureVersions(route *routev1.Route, reason, message string) {
-	performIngressConditionUpdate(unservableInFutureVersionsAction, a.lease, a.tracker, a.client, a.lister, route, a.routerName, a.routerCanonicalHostname, routev1.RouteIngressCondition{
+	expectedCondition := routev1.RouteIngressCondition{
 		Type:    routev1.RouteUnservableInFutureVersions,
 		Status:  corev1.ConditionTrue,
 		Reason:  reason,
 		Message: message,
-	})
+	}
+	// First, verify if updates are required by checking if the route ingress status matches expected values.
+	// This helps avoid unnecessary tasks in the writerlease queue and prevents unneeded lease extensions.
+	// We only do this in for the new UnservableInFutureVersions condition to avoid perturbing existing logic for the
+	// Admitted condition. This should be reevaluated when refactoring to use a single route status update per route
+	// reconciliation (see TODO at top of file).
+	changed, _, _, _, _ := recordIngressCondition(route.DeepCopy(), a.routerName, a.routerCanonicalHostname, expectedCondition)
```

Comment:
Please add a comment to note that `route.DeepCopy()` is necessary because `recordIngressCondition` actually modifies `route`.

Reply:
Consider refactoring to use `findCondition` instead of `recordIngressCondition` to save some overhead from deepcopies etc.

Reply:
Done.